### DISCUSSION TITLE
How test harness recognize DUT is border router through ethernet?

### DISCUSSION THREAD

**Discussion by ken-ke** [openthread/openthread#8900] [Q&A]
Hello,

   I'm working on integrate new IEEE802.15.4 chip to raspberry pi, and have problem on test harness tool to recognize my device as border router.

   I follow this page to setup border router on my raspberry pi  (https://openthread.io/guides/border-router/build), and observe otbr-agent is workable and register mesh service successfully like below log.

"Mar 20 16:40:25 raspberrypi otbr-agent[11524]: [INFO]-BA------: Result of publish meshcop service OpenThread BorderRouter #6F29._meshcop._udp.local: OK"

   But test harness can't recognize my raspberry pi as border router, I try capture Ethernet log and find test harness tool sent packet like below when doing "discovery network devices".

> Internet Protocol Version 6, Src: fe80::512:3b60:8b6f:61fc, Dst: ff02::114
> User Datagram Protocol, Src Port: 62693, Dst Port: 12345
> Data (3 bytes)
>     Data: 424252
>     [Length: 3]

  And golden border router (eg: OTNCS_BR) responses it. But my device won't, is anything else I need to setup?

> Internet Protocol Version 6, Src: fe80::125b:5c59:e059:9d5e, Dst: fe80::512:3b60:8b6f:61fc
> User Datagram Protocol, Src Port: 12345, Dst Port: 62693
> Data (99 bytes)
>     Data: 7b2276656e223a20224f544e43535f4252222c20226d6f64223a20224f544e43535f4252…

**Comment by jwhui** [comment]
Make sure you are setting `INFRA_IF_NAME` environment variable to `eth0`.

You may find the https://github.com/openthread/ot-reference-release repo helpful. The OTNCS_BR is built out of this repo.

**Reply by ken-ke** [reply]
Yes, I set `INFRA_IF_NAME` as `eth0`, I will try setup with ot-reference-release repo.

**Reply by ken-ke** [reply]
@jwhui  I found solution just need to run the process "testharness-discovery" to advertise raspberry pi 's IP address and device name.
However thanks for your support.

**Reply by jwhui** [reply]
Great to hear you figured it out. Note that `testharness-discovery` service should be enabled by default when using the OTBR install scripts.

https://github.com/openthread/ot-br-posix/blob/cbeaf817c54ad06d0238d0fa2ec2a5b4b32ba805/script/_otbr#L139
